test/sql/index/art/memory/test_art_varchar.test_slow

```diff
@@ -0,0 +1,80 @@
+# name: test/sql/index/art/memory/test_art_varchar.test_slow
+# description: Test the memory usage of the ART for a big table with a VARCHAR column
+# group: [memory]
+
+# test issue #7760
+
+require vector_size 2048
+
+statement ok
+PRAGMA enable_verification
+
+statement ok
+CREATE FUNCTION mem_to_bytes(x) AS CASE
+    WHEN CONTAINS(x, 'KB') THEN REPLACE(x, 'KB', '')::BIGINT * 1000
+    WHEN CONTAINS(x, 'MB') THEN REPLACE(x, 'MB', '')::BIGINT * 1000 * 1000
+    WHEN CONTAINS(x, 'GB') THEN REPLACE(x, 'GB', '')::BIGINT * 1000 * 1000 * 1000
+    WHEN CONTAINS(x, 'TB') THEN REPLACE(x, 'TB', '')::BIGINT * 1000 * 1000 * 1000 * 1000
+    WHEN x = '0 bytes' THEN 0::BIGINT
+    ELSE x::BIGINT END;
+
+# 7200000 unique strings
+
+statement ok
+CREATE TABLE strings AS
+    SELECT ((i * 95823983533) % 86000000)::VARCHAR AS s1
+        FROM range(7200000) tbl(i);
+
+statement ok
+CREATE TABLE art AS SELECT rpad(s1, 10, '-') AS id FROM strings;
+
+statement ok
+DROP TABLE strings;
+
+# 2 * 7200k entries
+statement ok
+INSERT INTO art (SELECT * FROM art);
+
+# 4 * 7200k entries
+statement ok
+INSERT INTO art (SELECT * FROM art);
+
+# 8 * 7200k entries
+statement ok
+INSERT INTO art (SELECT * FROM art);
+
+# 86M entries
+statement ok
+INSERT INTO art (SELECT * FROM art LIMIT 28400000);
+
+query I
+SELECT count(*) FROM art;
+----
+86000000
+
+query I
+SELECT COUNT(DISTINCT id) FROM art;
+----
+7200000
+
+query II
+SELECT MIN(length(id)), MAX(length(id)) FROM art;
+----
+10	10
+
+statement ok
+CREATE TABLE base AS
+SELECT mem_to_bytes(memory_usage)::BIGINT AS usage FROM pragma_database_size();
+
+statement ok
+SET memory_limit='12GB';
+
+statement ok
+CREATE INDEX idx ON art USING ART(id);
+
+query I
+SELECT mem_to_bytes(current.memory_usage) > base.usage AND
+       	mem_to_bytes(current.memory_usage) < 4 * base.usage
+FROM base, pragma_database_size() current;
+----
+1
```

Comment:
If I read this correctly:
This test only checks memory usage after index creation, and it can be anywhere from 1 to 4x the memory usage of just having the table, right?
Does the range need to be that wide? I would assume that for a memory representation when compared to uncompressed varchar columns it wouldn't blow up that much.

The other thing I'm thinking is if we can somehow test memory consumption during one operation, as in, when running the create index you don't get a sneaky memory usage spike
But wasn't the issue 
the index can increase the memory usage

Reply:
> This test only checks memory usage after index creation, and it can be anywhere from 1 to 4x the memory usage of just having the table, right?

Yes.

> Does the range need to be that wide? I would assume that for a memory representation when compared to uncompressed varchar columns it wouldn't blow up that much.

Nice catch, this is probably a relic. I will double-check and see if I can decrease the range.

> The other thing I'm thinking is if we can somehow test memory consumption during one operation, as in, when running the create index you don't get a sneaky memory usage spike

I added a benchmark with a fixed (low) memory limit. If we start spiking the memory usage again, that benchmark will hopefully be very slow due to offloading/ or even time-out. But now I'm thinking that it might be worth adding a few more of these benchmarks...

> But wasn't the issue
the index can increase the memory usage

How do you mean this?

Reply:
It does blow up quite a bit, from ~1 GB to ~3 GB. This benchmark has duplicates, so we cannot inline the leaves. Storing the column twice already gives us 2GB (10-byte strings vs 8-byte row IDs). Whatever we gain here from the prefix compression is probably countered by the memory of the nodes... And then we have another GB for the leaves or so. I can look more closely at where the memory goes, exactly (how many of each node and such). Should I? :)